examples/petstore-expanded/fiber/api/petstore-server.gen.go

```diff
@@ -0,0 +1,245 @@
+// Package api provides primitives to interact with the openapi HTTP API.
+//
+// Code generated by github.com/deepmap/oapi-codegen version (devel) DO NOT EDIT.
+package api
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/deepmap/oapi-codegen/pkg/runtime"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/gofiber/fiber/v2"
+)
+
+// ServerInterface represents all server handlers.
+type ServerInterface interface {
+	// Returns all pets
+	// (GET /pets)
+	FindPets(c *fiber.Ctx, params FindPetsParams) error
+	// Creates a new pet
+	// (POST /pets)
+	AddPet(c *fiber.Ctx) error
+	// Deletes a pet by ID
+	// (DELETE /pets/{id})
+	DeletePet(c *fiber.Ctx, id int64) error
+	// Returns a pet by ID
+	// (GET /pets/{id})
+	FindPetByID(c *fiber.Ctx, id int64) error
+}
+
+// ServerInterfaceWrapper converts contexts to parameters.
+type ServerInterfaceWrapper struct {
+	Handler ServerInterface
+}
+
+type MiddlewareFunc fiber.Handler
+
+// FindPets operation middleware
+func (siw *ServerInterfaceWrapper) FindPets(c *fiber.Ctx) error {
+
+	var err error
+
+	// Parameter object where we will unmarshal all parameters from the context
+	var params FindPetsParams
+
+	query, err := url.ParseQuery(string(c.Request().URI().QueryString()))
```

Comment:
This could use fiber's `func (c *Ctx) QueryParser(out interface{}) error` func https://docs.gofiber.io/api/ctx/#queryparser

Reply:
After some investigation it seems like this requires all the generated request structs to be tagged with `query: <name>`. 

Gonna leave it as a future improvement, since it looks like it will require changes to the type generation potentially affecting the other server implementations too.